test/systemtests/aci_test.go

```diff
@@ -72,403 +73,582 @@ func (s *systemtestSuite) TestACIMode(c *C) {
-		TenantName:  "aciTenant",
-		NetworkName: "aciNet",
+		TenantName:  s.globInfo.Tenant,
+		NetworkName: s.globInfo.Network,
 		GroupName:   "epga",
 	}), IsNil)
 
-	c.Assert(s.cli.AppProfilePost(&client.AppProfile{
-		TenantName:     "aciTenant",
-		EndpointGroups: []string{"epga"},
-		AppProfileName: "profile1",
-	}), IsNil)
-
-	containersA, err := s.runContainersOnNode(1, "aciNet", "aciTenant", "epga", s.nodes[0])
+	err = s.nodes[0].exec.checkSchedulerNetworkCreated("epga", true)
 	c.Assert(err, IsNil)
 
-	// Verify cA1 can ping default gateway
-	c.Assert(s.pingTestToNonContainer(containersA, []string{"20.1.1.254"}), IsNil)
+	for i := 0; i < s.basicInfo.Iterations; i++ {
+		c.Assert(s.cli.AppProfilePost(&client.AppProfile{
+			TenantName:     s.globInfo.Tenant,
+			EndpointGroups: []string{"epga"},
+			AppProfileName: "profile1",
+		}), IsNil)
+		time.Sleep(5 * time.Second)
+
+		containers, err := s.runContainersInGroups(s.basicInfo.Containers, s.globInfo.Network, s.globInfo.Tenant, []string{"epga"})
+		c.Assert(err, IsNil)
 
-	c.Assert(s.removeContainers(containersA), IsNil)
-	c.Assert(s.cli.AppProfileDelete("aciTenant", "profile1"), IsNil)
-	c.Assert(s.cli.EndpointGroupDelete("aciTenant", "epga"), IsNil)
-	c.Assert(s.cli.NetworkDelete("aciTenant", "aciNet"), IsNil)
-}
+		for key := range containers {
+			containersA = append(containersA, key)
+		}
+
+		// Verify containers in A can ping default gateway
+		c.Assert(s.pingTestByName(containersA, s.globInfo.Gateway), IsNil)
 
+		c.Assert(s.removeContainers(containersA), IsNil)
+		containersA = nil
+		c.Assert(s.cli.AppProfileDelete(s.globInfo.Tenant, "profile1"), IsNil)
+		time.Sleep(time.Second * 5)
+	}
+	c.Assert(s.cli.EndpointGroupDelete(s.globInfo.Tenant, "epga"), IsNil)
+	c.Assert(s.cli.NetworkDelete(s.globInfo.Tenant, s.globInfo.Network), IsNil)
+}
 
+/* TestACIProfile does the following:
+1). TestACIMode and TestACIPingGateway for containers in two EPGs.
+2). Checks policies and rules learned on the APIC controller.
+3). Checks policies behavior before and after deleting app-profiles.
+*/
 func (s *systemtestSuite) TestACIProfile(c *C) {
-	if s.fwdMode == "routing" {
+	if s.fwdMode == "routing" || s.basicInfo.Scheduler == "k8" {
 		return
 	}
 	c.Assert(s.cli.GlobalPost(&client.Global{
 		Name:             "global",
 		NetworkInfraType: "aci",
-		Vlans:            "1120-1200",
-		Vxlans:           "1-10000",
 		FwdMode:          "bridge",
+		Vlans:            s.globInfo.Vlan,
+		Vxlans:           s.globInfo.Vxlan,
 	}), IsNil)
 	c.Assert(s.cli.TenantPost(&client.Tenant{
-		TenantName: "aciTenant",
+		TenantName: s.globInfo.Tenant,
 	}), IsNil)
 
-	for i := 0; i < 2; i++ {
-		log.Infof(">>ITERATION #%d<<", i)
+	containersA := []*container{}
+	containersB := []*container{}
+
+	containersA2 := []*container{}
+	containersB2 := []*container{}
+	for i := 0; i < s.basicInfo.Iterations; i++ {
+
 		c.Assert(s.cli.NetworkPost(&client.Network{
```

Comment:
i am assuming the config file will have option to include multiple networks. This may need to be handled here looping over the number of networks listed in the config file. This can be part of separate PR. Please open a bug once the PR is merged